### PR TITLE
Tracing: Trace volume list operation

### LINKED ISSUE #1196: Tracing: Trace "volume list" command.
Add instrumentation to trace the "volume list" command. Also add attribute to the high level span on
the number of volumes being listed to help debugging latency issues (if any) during the listing
operation.

### Observed behavior
NA

### Expected/desired behavior
Tracing UI should show the high level steps and the time taken for each of the steps. The top level
span must also show the number of volumes being listed as part of the request on the tracing UI.

### Details on how to reproduce (minimal and precise)
NA

### Information about the environment:

- Glusterd2 version used (e.g. v4.1.0 or master): master
- Operating system used: fedora 27
- Glusterd2 compiled from sources, as a package (rpm/deb), or container: source
- Using External ETCD: (yes/no, if yes ETCD version): no
- If container, which container image: no
- Using kubernetes, openshift, or direct install: direct
- If kubernetes/openshift, is gluster running inside kubernetes/openshift or outside: 

### Other useful information

- glusterd2 config files from all nodes (default /etc/glusterd2/glusterd2.toml)
- glusterd2 log files from all nodes (default /var/log/glusterd2/glusterd2.log)
- ETCD configuration
- Contents of `uuid.toml` from all nodes (default /var/lib/glusterd2/uuid.toml)
- Output of `statedump` from any one of the node

### Useful commands

- To get glusterd2 version
    ```
    glusterd2 --version
    ```
- To get ETCD version
    ```
    etcd --version
    ```
- To get output of statedump
    ```
    curl http://glusterd2-IP:glusterd2-Port/statedump
    ```

### PR DESCRIPTION
**Pull request by sseshasa**
Add instrumentation to trace volume list operation. Add the count of the
number of volumes listed by the command as an attribute in the top level
span of the trace.

Closes #1196

### CODE REVIEW

File: glusterd2/commands/volumes/volume-create.go [hunk 1]
```diff
@@ -100,7 +101,12 @@ func registerVolCreateStepFuncs() {
 func volumeCreateHandler(w http.ResponseWriter, r *http.Request) {
 
 	ctx := r.Context()
-	ctx, span := trace.StartSpan(ctx, "/volumeCreateHandler")
+	fnName := "?"
+	if pc, _, _, ok := runtime.Caller(0); ok {
```
Comment (Madhu-1) [inline glusterd2/commands/volumes/volume-create.go]: now we can move this function to utils and use it

**Comment by sseshasa** [comment]
@Madhu-1 I addressed your comment. Please verify.

File: glusterd2/volume/store-utils.go [hunk 2]
```diff
@@ -7,10 +7,13 @@ import (
 	"github.com/gluster/glusterd2/glusterd2/brick"
 	"github.com/gluster/glusterd2/glusterd2/store"
 	gderror "github.com/gluster/glusterd2/pkg/errors"
+	"github.com/gluster/glusterd2/pkg/utils"
 
 	"github.com/coreos/etcd/clientv3"
 	"github.com/pborman/uuid"
 	log "github.com/sirupsen/logrus"
+
```
Comment (Madhu-1) [inline glusterd2/volume/store-utils.go]: no need of extra line, as these packages are third-party libraries

Comment (sseshasa) [reply glusterd2/volume/store-utils.go]: Sure, will remove it.

File: glusterd2/commands/volumes/volume-create.go [hunk 3]
```diff
@@ -100,7 +101,8 @@ func registerVolCreateStepFuncs() {
 func volumeCreateHandler(w http.ResponseWriter, r *http.Request) {
 
 	ctx := r.Context()
-	ctx, span := trace.StartSpan(ctx, "/volumeCreateHandler")
+	fnName := utils.GetFuncName()
+	ctx, span := trace.StartSpan(ctx, "/"+fnName)
```
Comment (kshlm) [inline glusterd2/commands/volumes/volume-create.go]: You really don't need to get the function name programatically here. You'd need the helper utility if `StartSpan` could be called dynamically at runtime, and you wouldn't know the callers before hand. But here you know the caller,  so you could just hard code the function name string whenever you call it. Just a simple `ctx, span := trace.StartSpan(ctx, "/volumeCreateHandler")` is good here. 
The same also applies to the other `StartSpan` calls.

Comment (sseshasa) [reply glusterd2/commands/volumes/volume-create.go]: I will revert it here and hard code it in the other places.

**Comment by prashanthpai** [comment]
@sseshasa Volume listing operation doesn't really need much of tracing. It's best if tracing is first added to those operations that create a transaction.

**Pull request merged**